storage-proofs/porep/src/stacked/vanilla/cores.rs

```diff
@@ -42,7 +42,25 @@ pub fn checkout_core_group() -> Option<MutexGuard<'static, CoreGroup>> {
     }
 }
 
-pub fn bind_core(core_index: CoreIndex) -> Result<()> {
+pub struct Cleanup {
+    prior_state: Option<hwloc::Bitmap>,
+}
+
+impl Drop for Cleanup {
+    fn drop(&mut self) {
+        match self.prior_state.take() {
+            Some(prior) => {
+                let child_topo = &TOPOLOGY;
+                let tid = get_thread_id();
```

Comment:
maybe store the thread id too, to be sure this does it for the correct thread?

Reply:
Done, thanks.